### PR TITLE
quincy: mgr/cephadm: fixups for asyncio based timeout

### PR DESCRIPTION
**Pull request by adk3798**
Backport of https://github.com/ceph/ceph/pull/51963

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: qa/workunits/cephadm/test_cephadm_timeout.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                           check=True, stdout=f_stdout, stderr=f_stderr)
+
+    res_stdout = open('/tmp/ceph-health-detail-stdout', 'r').read()
+    res_stderr = open('/tmp/ceph-health-detail-stderr', 'r').read()
+    print(f'"cephadm shell -- ceph health detail" stdout:\n{res_stdout}')
+    print(f'"cephadm shell -- ceph health detail" stderr:\n{res_stderr}')
+
+    print('Checking for correct health warning in health detail...')
+    if 'CEPHADM_REFRESH_FAILED' not in res_stdout:
+        raise Exception('No health warning caused by timeout was raised')
```
Comment (adk3798) [inline qa/workunits/cephadm/test_cephadm_timeout.py:167]: This test will fail in cases where the agent is on. There was a patch in main to make it so certain workunits, including this one, don't run with the agent. I guess we'll need that in quincy as well.

https://pulpito.ceph.com/adking-2024-03-06_13:11:39-orch:cephadm-wip-adk2-testing-2024-03-05-1437-quincy-distro-default-smithi/7583237/

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-03-06_13:11:39-orch:cephadm-wip-adk2-testing-2024-03-05-1437-quincy-distro-default-smithi/

only failure related to PRs in the run are the upgrade tests failing trying to run `ceph orch apply rgw smpl` as the image it's upgrading from (15.2.0) doesn't allow applying an rgw service without a zone specified. That PR will need to be adjusted, but the other ones in the run should be okay.

**Comment by phlogistonjohn** [comment]
jenkins test make check

**Review by phlogistonjohn — APPROVED** [review]

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
jenkins test dashboard cephadm

**Pull request merged**